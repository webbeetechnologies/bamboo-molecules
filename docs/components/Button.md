# Button
- As a developer, I want to inject the base Button (React-native replacement) component.
- As a user, I want to see feedback from button on hover and focus.
- As a user, I want to know that the button is disabled.
- As a developer, I want to have Icons on left and/or right of the button.

## Implementation Details
- Implements [WithElements HOC](./HOC/WithElements.md)

## References
- [Material 3 Figma Design](https://www.figma.com/file/hGlWM0wnP1UN5VtPGiaOJ2/Material-3-Design-Kit-(Community)?node-id=50716%3A11360)
- [Material 3 Design Guidelines](https://m3.material.io/components/buttons/overview)

## Screenshots
#### Material 3 Button
![Material 3 Buttons](./assets/m3.buttons.png)

# Button Group
- As a developer, I want the ability to group buttons;

## Implementation Details
- Implements [WithAddons HOC](./HOC/WithAddons.md)
- Implements [WithRipple HOC](./HOC/WithRipple.md)

## References
- [Material 3 Segmented Buttons](https://m3.material.io/components/segmented-buttons/overview)

## Screenshots
#### Material 3 Segmented Button
![Material 3 Segmented Buttons](./assets/m3.button-groups.png)