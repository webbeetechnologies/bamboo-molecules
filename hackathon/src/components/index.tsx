export { default as AttributeValueInput } from './AttributeValueInput/AttributeValueInput';
export { default as CustomDrawer } from './CustomDrawer/CustomDrawer';
export { default as MachineCard } from './MachineCard/MachineCard';
export { default as MachineTypeCard } from './MachineTypeCard/MachineTypeCard';
