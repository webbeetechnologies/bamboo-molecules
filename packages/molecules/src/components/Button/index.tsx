export { default as Button, Props as ButtonProps } from './Button';
export { defaultStyles as buttonStyles } from './utils';
export * from './types';
