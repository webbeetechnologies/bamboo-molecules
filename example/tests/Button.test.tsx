import { Button } from '../../src/components';
import { Contained as ButtonStory } from '../stories/components/Button/Button.stories';
import type { Props } from '../stories/components/Button/Button';
import { renderWithWrapper, fireEvent, runBenchmark } from '../testHelpers';

describe('button', () => {
    it('renders text button by default', () => {
        const tree = renderWithWrapper(<Button>Text Button</Button>).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders text button with variant', () => {
        const tree = renderWithWrapper(<Button variant="text">Text Button</Button>).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders outlined button with variant', () => {
        const tree = renderWithWrapper(
            <Button variant="outlined">Outlined Button</Button>,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders contained contained with variant', () => {
        const tree = renderWithWrapper(
            <Button variant="contained">Contained Button</Button>,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders loading button', () => {
        const tree = renderWithWrapper(<Button loading>Loading Button</Button>).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders disabled button', () => {
        const tree = renderWithWrapper(<Button disabled>Disabled Button</Button>).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders button with color', () => {
        const tree = renderWithWrapper(
            <Button textColor="colors.tertiary">Custom Button</Button>,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders button with custom background color', () => {
        const tree = renderWithWrapper(
            <Button buttonColor="colors.tertiary">Custom Button</Button>,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders button with custom testID', () => {
        const tree = renderWithWrapper(
            <Button testID={'custom:testID'}>Button with custom testID</Button>,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders button with an accessibility label', () => {
        const tree = renderWithWrapper(
            <Button accessibilityLabel={'label'}>Button with accessibility label</Button>,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('renders button with an accessibility hint', () => {
        const tree = renderWithWrapper(
            <Button accessibilityHint={'hint'}>Button with accessibility hint</Button>,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should execute onPressIn', () => {
        const onPressInMock = jest.fn();
        const onPress = jest.fn();

        const { getByTestId } = renderWithWrapper(
            <Button onPress={onPress} onPressIn={onPressInMock} testID="button">
                Button with onPressIn
            </Button>,
        );
        fireEvent(getByTestId('button'), 'onPressIn');
        expect(onPressInMock).toHaveBeenCalledTimes(1);
    });

    it('should execute onPressOut', () => {
        const onPressOutMock = jest.fn();
        const onPress = jest.fn();

        const { getByTestId } = renderWithWrapper(
            <Button onPress={onPress} onPressOut={onPressOutMock} testID="button">
                Button with onPressIn
            </Button>,
        );
        fireEvent(getByTestId('button'), 'onPressOut');
        expect(onPressOutMock).toHaveBeenCalledTimes(1);
    });

    it.only('benchmark button', done => {
        console.log('benchmark start');
        const results: any = Promise.race([
            runBenchmark({
                component: ButtonStory,
                props: ButtonStory.args as Props,
                timeout: 500,
            }),
            new Promise((x, reject) => setTimeout(reject, 600)),
        ]).then(() => {
            console.log('benchmark done');
            expect(results?.mean).toBeLessThan(4);
            done();
        });
    });
});
