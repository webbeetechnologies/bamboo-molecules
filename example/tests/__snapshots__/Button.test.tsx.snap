// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button renders button with an accessibility hint 1`] = `
<View>
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": undefined,
        "bottom": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        focused={false}
        pressed={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 750,
            "borderStyle": "solid",
            "flex": 1,
            "minWidth": 64,
          }
        }
      >
        <View
          accessibilityHint="hint"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 750,
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Object {
                  "alignItems": "center",
                  "color": "",
                  "display": "flex",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                  "marginHorizontal": 12,
                  "marginVertical": 10,
                  "textAlign": "center",
                }
              }
            >
              Button with accessibility hint
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`button renders button with an accessibility label 1`] = `
<View>
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": undefined,
        "bottom": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        focused={false}
        pressed={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 750,
            "borderStyle": "solid",
            "flex": 1,
            "minWidth": 64,
          }
        }
      >
        <View
          accessibilityLabel="label"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 750,
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Object {
                  "alignItems": "center",
                  "color": "",
                  "display": "flex",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                  "marginHorizontal": 12,
                  "marginVertical": 10,
                  "textAlign": "center",
                }
              }
            >
              Button with accessibility label
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`button renders button with color 1`] = `
<View>
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": undefined,
        "bottom": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        focused={false}
        pressed={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 750,
            "borderStyle": "solid",
            "flex": 1,
            "minWidth": 64,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 750,
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Object {
                  "alignItems": "center",
                  "color": "",
                  "display": "flex",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                  "marginHorizontal": 12,
                  "marginVertical": 10,
                  "textAlign": "center",
                }
              }
            >
              Custom Button
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`button renders button with custom background color 1`] = `
<View>
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": undefined,
        "bottom": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        focused={false}
        pressed={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 750,
            "borderStyle": "solid",
            "flex": 1,
            "minWidth": 64,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 750,
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Object {
                  "alignItems": "center",
                  "color": "",
                  "display": "flex",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                  "marginHorizontal": 12,
                  "marginVertical": 10,
                  "textAlign": "center",
                }
              }
            >
              Custom Button
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`button renders button with custom testID 1`] = `
<View>
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": undefined,
        "bottom": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        focused={false}
        pressed={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 750,
            "borderStyle": "solid",
            "flex": 1,
            "minWidth": 64,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 750,
                "flex": 1,
              },
            ]
          }
          testID="custom:testID"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Object {
                  "alignItems": "center",
                  "color": "",
                  "display": "flex",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                  "marginHorizontal": 12,
                  "marginVertical": 10,
                  "textAlign": "center",
                }
              }
            >
              Button with custom testID
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`button renders contained contained with variant 1`] = `
<View>
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": undefined,
        "bottom": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        focused={false}
        pressed={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 750,
            "borderStyle": "solid",
            "flex": 1,
            "minWidth": 64,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 750,
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Object {
                  "alignItems": "center",
                  "color": "",
                  "display": "flex",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                  "marginHorizontal": 24,
                  "marginVertical": 10,
                  "textAlign": "center",
                }
              }
            >
              Contained Button
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`button renders disabled button 1`] = `
<View>
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": undefined,
        "bottom": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        focused={false}
        pressed={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 750,
            "borderStyle": "solid",
            "flex": 1,
            "minWidth": 64,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 750,
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Object {
                  "alignItems": "center",
                  "color": "",
                  "display": "flex",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                  "marginHorizontal": 12,
                  "marginVertical": 10,
                  "textAlign": "center",
                }
              }
            >
              Disabled Button
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`button renders loading button 1`] = `
<View>
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": undefined,
        "bottom": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        focused={false}
        pressed={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 750,
            "borderStyle": "solid",
            "flex": 1,
            "minWidth": 64,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 750,
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <View
              accessibilityRole="progressbar"
              accessibilityState={
                Object {
                  "busy": true,
                }
              }
              accessible={true}
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginLeft": 12,
                  "marginRight": -8,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 26,
                    "opacity": 1,
                    "width": 26,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 26,
                        "transform": Array [
                          Object {
                            "rotate": "45deg",
                          },
                        ],
                        "width": 26,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "height": 13,
                          "overflow": "hidden",
                          "width": 26,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "height": 26,
                            "transform": Array [
                              Object {
                                "translateY": 0,
                              },
                              Object {
                                "rotate": "-165deg",
                              },
                            ],
                            "width": 26,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "height": 13,
                              "overflow": "hidden",
                              "width": 26,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              Object {
                                "borderColor": "rgba(103, 80, 164, 1)",
                                "borderRadius": 13,
                                "borderWidth": 2.6,
                                "height": 26,
                                "width": 26,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  collapsable={false}
                  style={
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 26,
                        "transform": Array [
                          Object {
                            "rotate": "45deg",
                          },
                        ],
                        "width": 26,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "height": 13,
                          "overflow": "hidden",
                          "top": 13,
                          "width": 26,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "height": 26,
                            "transform": Array [
                              Object {
                                "translateY": -13,
                              },
                              Object {
                                "rotate": "345deg",
                              },
                            ],
                            "width": 26,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "height": 13,
                              "overflow": "hidden",
                              "width": 26,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              Object {
                                "borderColor": "rgba(103, 80, 164, 1)",
                                "borderRadius": 13,
                                "borderWidth": 2.6,
                                "height": 26,
                                "width": 26,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Object {
                  "alignItems": "center",
                  "color": "",
                  "display": "flex",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                  "marginHorizontal": 16,
                  "marginVertical": 10,
                  "textAlign": "center",
                }
              }
            >
              Loading Button
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`button renders outlined button with variant 1`] = `
<View>
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": undefined,
        "bottom": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        focused={false}
        pressed={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "",
            "borderRadius": 750,
            "borderStyle": "solid",
            "borderWidth": 1,
            "flex": 1,
            "minWidth": 64,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 750,
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Object {
                  "alignItems": "center",
                  "color": "",
                  "display": "flex",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                  "marginHorizontal": 24,
                  "marginVertical": 10,
                  "textAlign": "center",
                }
              }
            >
              Outlined Button
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`button renders text button by default 1`] = `
<View>
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": undefined,
        "bottom": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        focused={false}
        pressed={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 750,
            "borderStyle": "solid",
            "flex": 1,
            "minWidth": 64,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 750,
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Object {
                  "alignItems": "center",
                  "color": "",
                  "display": "flex",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                  "marginHorizontal": 12,
                  "marginVertical": 10,
                  "textAlign": "center",
                }
              }
            >
              Text Button
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`button renders text button with variant 1`] = `
<View>
  <View
    collapsable={false}
    style={
      Object {
        "alignSelf": undefined,
        "bottom": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "top": undefined,
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        focused={false}
        pressed={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 750,
            "borderStyle": "solid",
            "flex": 1,
            "minWidth": 64,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 750,
                "flex": 1,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Object {
                  "alignItems": "center",
                  "color": "",
                  "display": "flex",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                  "marginHorizontal": 12,
                  "marginVertical": 10,
                  "textAlign": "center",
                }
              }
            >
              Text Button
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
